File: InclusiON.DTOs/Responses/Diagnoses/DiagnosisResponse.cs
using InclusiON.Domain.Models;

namespace InclusiON.DTOs.Responses.Diagnoses
{
    public class DiagnosisResponse
    {
        public int Id { get; set; }
        public Guid PersonId { get; set; }
        public Guid ProfessionalId { get; set; }
        public string ProfessionalName { get; set; } = string.Empty;
        public DateTime DiagnosisDate { get; set; }
        public string PrimaryDiagnosis { get; set; } = string.Empty;
        public string? InitialObservations { get; set; }
        public string? IdentifiedCapabilities { get; set; }
        public string? IdentifiedChallenges { get; set; }
        public string? RequiredSupports { get; set; }
        public string? PedagogicalObjectives { get; set; }
        public string? RecommendedStrategies { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        public static DiagnosisResponse MapToResponse(Diagnosis d)
        {
            return new DiagnosisResponse
            {
                Id = d.Id,
                PersonId = d.PersonId,
                ProfessionalId = d.ProfessionalId,
                ProfessionalName = $"{d.Professional.FirstName} {d.Professional.LastName}".Trim(),
                DiagnosisDate = d.DiagnosisDate,
                PrimaryDiagnosis = d.PrimaryDiagnosis,
                InitialObservations = d.InitialObservations,
                IdentifiedCapabilities = d.IdentifiedCapabilities,
                IdentifiedChallenges = d.IdentifiedChallenges,
                RequiredSupports = d.RequiredSupports,
                PedagogicalObjectives = d.PedagogicalObjectives,
                RecommendedStrategies = d.RecommendedStrategies,
                CreatedAt = d.CreatedAt,
                UpdatedAt = d.UpdatedAt
            };
        }
    }
}

File: src/app/models/responses/diagnosis.response.ts
export interface DiagnosisResponse {
  id: number;
  personId: string;
  professionalId: string;
  professionalName: string;
  diagnosisDate: string;
  primaryDiagnosis: string;
  initialObservations?: string;
  identifiedCapabilities?: string;
  identifiedChallenges?: string;
  requiredSupports?: string;
  pedagogicalObjectives?: string;
  recommendedStrategies?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface DiagnosisListItemResponse {
  id: number;
  diagnosisDate: string;
  primaryDiagnosis: string;
  professionalName: string;
  professionalId: string;
  createdByUserId: string;
  createdAt: string;
}
